otx/algorithms/segmentation/adapters/mmseg/configurer.py

```diff
@@ -296,11 +296,11 @@ def configure_classes(self, cfg: Config) -> None:
         # Model architecture
         if "decode_head" in cfg.model:
             decode_head = cfg.model.decode_head
-            if isinstance(decode_head, Config):
-                decode_head.num_classes = len(model_classes)
-            elif isinstance(decode_head, list):
+            if isinstance(decode_head, list):
                 for head in decode_head:
                     head.num_classes = len(model_classes)
+            else:
```

Comment:
What about inspecting `Config` and `ConfigDict` together?
`elif isinstance(decode_head, (Config, ConfigDict))`

Reply:
I think it's okay to leave as is, because it is not inspecting in the list if-statement as well assuming that decode_head is dict type.